File: commands/find.py
from telegram import InlineKeyboardButton, InlineKeyboardMarkup

from utils.gettorrentsjson import returnResult


def search_query(bot, update, args):
    search_query = ' '.join(args)
    chat_id = update.message.chat.id
    message_id = update.message.message_id
    json = returnResult(search_query)
    if len(json) == 0:
        bot.sendSticker(chat_id, 'CAADBQADYQADZ7RFFii2rILMAAG4cQI',
                        reply_to_message_id=update.message.message_id)
        return
    query_data = ""
    keyboard = list()
    if len(json) >= 20:
        limit = 20
    else:
        limit = len(json)
    for i in range(limit):
        keyboard.append([
            InlineKeyboardButton(json[i]['title'],
                                 callback_data=i)
        ])

    reply_markup = InlineKeyboardMarkup(keyboard)
    message = bot.sendMessage(chat_id, "<b>🔍Search Query Results:👇</b>\n"
                                       "<pre>Search Query : </pre>" + "<i>" + search_query + "</i>\n",
                              parse_mode='HTML',
                              reply_markup=reply_markup,
                              reply_to_message_id=update.message.message_id,
                              disable_web_page_preview=True)

File: commands/find.js
const config = require("../config");
const telebot = require("telebot");
const flatCache = require("flat-cache");
const search = require("../utils/api");
const path = require('path')
const bot = new telebot(config.botToken);
const parseMode = config.parseMode;

var cachePath = path.resolve(__dirname, "../cache");
var cache = flatCache.load("itorrent", cachePath);

const find = async (msg, props) => {
  const searchQuery = props.match[1];
  const buttons = [];
  const searchJson = await search(searchQuery);
  if (searchJson.length <= 0) {
    return msg.reply.text("Oops! Couldnt Find Anything");
  }
  const queryKey = Math.random().toString(36).substring(7);
  cache.setKey(queryKey, searchJson);
  for (var i = 0; i <= 20; i++) {
    const callback = Math.random().toString(36).substring(7) + ":" + queryKey;
    cache.setKey(callback, searchJson[i]);
    buttons.push([
      bot.inlineButton(searchJson[i].name, {
        callback,
      }),
    ]);
  }
  const replyMarkup = bot.inlineKeyboard(buttons);
  cache.save(true);
  return bot
    .sendMessage(msg.chat.id, "<b>🔍Search Query Results:👇</b>\n", {
      parseMode,
      replyMarkup,
      notification: true,
      replyToMessage: msg.message_id,
      webPreview: false,
    })
    .catch((err) => console.log(err));
};
module.exports = find;
